entity/contents/filters/filter_input.h

```diff
@@ -0,0 +1,51 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#pragma once
+
+#include <memory>
+#include <optional>
+#include <variant>
+#include <vector>
+
+#include "impeller/entity/contents/contents.h"
+#include "impeller/geometry/rect.h"
+#include "impeller/renderer/formats.h"
+
+namespace impeller {
+
+class ContentContext;
+class Entity;
+
+class FilterInput final {
```

Comment:
This is extremely cool and elegant. Especially when introducing a new concept though, please document the class. What you wrote in the PR description would be a great start.

Reply:
Curious: Why final by default? Though, TBH that should be the default throughout. Prevents folks from mocking these.

Reply:
Done -- added a docstring for the class.

Reply:
Oh, the `final` was just compulsory -- I can remove it if you'd like.